### ISSUE TITLE
integration does not load on 2024.11.0b0

### ISSUE THREAD

**Issue by ChristophCaina** [hg1337/homeassistant-dwd#32]
The integration can't load on HA 2024.11.0b0 and throws the following error:

```
Logger: homeassistant.setup
Quelle: setup.py:334
Erstmals aufgetreten: 09:03:55 (1 Vorkommnisse)
Zuletzt protokolliert: 09:03:55

Setup failed for custom integration 'dwd': Unable to import component: cannot import name 'Config' from 'homeassistant.core' (/usr/src/homeassistant/homeassistant/core.py)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1011, in async_get_component
    comp = await self.hass.async_add_import_executor_job(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1071, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/dwd/__init__.py", line 16, in <module>
    from homeassistant.core import Config, HomeAssistant
ImportError: cannot import name 'Config' from 'homeassistant.core' (/usr/src/homeassistant/homeassistant/core.py). Did you mean: 'config'?

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component
    component = await integration.async_get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1031, in async_get_component
    self._component_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1023, in async_get_component
    comp = self._get_component()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1071, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/dwd/__init__.py", line 16, in <module>
    from homeassistant.core import Config, HomeAssistant
ImportError: cannot import name 'Config' from 'homeassistant.core' (/usr/src/homeassistant/homeassistant/core.py). Did you mean: 'config'?
```

**Comment by kersten-lohmeyer** [comment]
Same here

**Comment by hg1337** [comment]
Thank you for reporting, I'm going to provide a fix tomorrow.

**Comment by hg1337** [comment]
> EDIT: Fixed with 2.1.21

I believe you are talking about the other integration at https://github.com/FL550/dwd_weather

**Comment by kersten-lohmeyer** [comment]
> > EDIT: Fixed with 2.1.21
> 
> I believe you are talking about the other integration at https://github.com/FL550/dwd_weather

Oh, ok, I might have mixed that up.

**Comment by hg1337** [comment]
> > > EDIT: Fixed with 2.1.21
> 
> > 
> 
> > I believe you are talking about the other integration at https://github.com/FL550/dwd_weather
> 
> 
> 
> Oh, ok, I might have mixed that up. But still I had a very similar error.

Probably many/all custom integrations have this problem because Home Assistant introduced a breaking change. Seems that they are new doing a soft transition: https://github.com/home-assistant/core/pull/129537

**Comment by hg1337** [comment]
In Home Assistant 2024.11.0b1 you should now get only a warning instead of an error. The warning is resolved with the latest release [2024.11.0](https://github.com/hg1337/homeassistant-dwd/releases/tag/2024.11.0) of this integration.
